platform-sdk/swirlds-config-processor/src/main/java/com/swirlds/config/processor/ConstantClassFactory.java

```diff
@@ -0,0 +1,92 @@
+
+public class ConstantClassFactory {
+
+    private ConstantClassFactory() {}
+
+    public static void doWork(
+            @NonNull final ConfigDataRecordDefinition configDataRecordDefinition,
+            @NonNull final JavaFileObject constantsSourceFile)
+            throws IOException {
+
+        Objects.requireNonNull(configDataRecordDefinition, "configDataRecordDefinition must not be null");
+        Objects.requireNonNull(constantsSourceFile, "constantsSourceFile must not be null");
+
+        final String orginalRecordClassName =
+                configDataRecordDefinition.packageName() + "." + configDataRecordDefinition.simpleClassName();
+        final JavaClassSource javaClassSource = Roaster.create(JavaClassSource.class);
+        javaClassSource.setPackage(configDataRecordDefinition.packageName());
+        javaClassSource.setName(
+                configDataRecordDefinition.simpleClassName() + ConfigProcessorConstants.CONSTANTS_CLASS_SUFFIX);
+        javaClassSource.setFinal(true);
+        javaClassSource
+                .getJavaDoc()
+                .setText("Constaints constants for all the property names that are part of {@link "
+                        + orginalRecordClassName + "}. Generated by {@code " + ConstantClassFactory.class.getName()
+                        + "} on "
+                        + DateTimeFormatter.ISO_DATE_TIME.format(LocalDateTime.now()) + ".");
+        javaClassSource.getJavaDoc().addTagValue("@see", orginalRecordClassName);
+
+        configDataRecordDefinition.propertyDefinitions().forEach(propertyDefinition -> {
+            final String name = toConstantName(
+                    propertyDefinition.name().replace(configDataRecordDefinition.configDataName() + ".", ""));
+            javaClassSource
+                    .addField()
+                    .setName(name)
+                    .setType(ConfigProcessorConstants.JAVA_LANG_STRING)
+                    .setLiteralInitializer("\"" + propertyDefinition.name() + "\"")
+                    .setPublic()
+                    .setStatic(true)
+                    .getJavaDoc()
+                    .setText("Name of the {@link " + orginalRecordClassName + "#" + propertyDefinition.fieldName()
+                            + "} property")
+                    .addTagValue("@see", orginalRecordClassName + "#" + propertyDefinition.fieldName());
+        });
+        try (final Writer writer = constantsSourceFile.openWriter()) {
+            writer.append(javaClassSource.toString());
+        }
+    }
+
+    @NonNull
+    public static String toConstantName(@NonNull final String propertyName) {
+        Objects.requireNonNull(propertyName, "propertyName must not be null");
+
+        final StringBuilder builder = new StringBuilder();
+        for (int i = 0; i < propertyName.length(); i++) {
+            final char character = propertyName.charAt(i);
+            if (Character.isUpperCase(character)) {
```

Comment:
```suggestion
            if (i > 0 && Character.isUpperCase(character)) {
```
To prevent some random names starts with uppercase